diff --git a/node_modules/@nx/js/src/utils/package-json/update-package-json.js b/node_modules/@nx/js/src/utils/package-json/update-package-json.js
index 0d47ce7..fdaed5a 100644
--- a/node_modules/@nx/js/src/utils/package-json/update-package-json.js
+++ b/node_modules/@nx/js/src/utils/package-json/update-package-json.js
@@ -39,6 +39,15 @@ function updatePackageJson(options, context, target, dependencies, fileMap = nul
     }
     // update package specific settings
     packageJson = getUpdatedPackageJsonContent(packageJson, options);
+
+    {
+      Object.keys(packageJson.dependencies).forEach((key) => {
+        if(options?.excludeDeps?.includes(key)) {
+          delete packageJson.dependencies[key]
+        }
+      })
+    }
+
     // save files
     (0, devkit_1.writeJsonFile)(`${options.outputPath}/package.json`, packageJson);
     if (options.generateLockfile) {
